crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -301,6 +306,82 @@ impl CheckpointStore {
         self.full_checkpoint_content.get(&seq)
     }
 
+    fn prune_local_summaries(&self) -> SuiResult {
+        if let Some((last_local_summary, _)) = self
+            .locally_computed_checkpoints
+            .unbounded_iter()
+            .skip_to_last()
+            .next()
+        {
+            let mut batch = self.locally_computed_checkpoints.batch();
+            batch.delete_range(&self.locally_computed_checkpoints, &0, &last_local_summary)?;
+            batch.write()?;
+            info!("Pruned local summaries up to {:?}", last_local_summary);
+        }
+        Ok(())
+    }
+
+    fn check_for_checkpoint_fork(
+        &self,
+        local_checkpoint: &CheckpointSummary,
+        verified_checkpoint: &VerifiedCheckpoint,
+    ) {
+        if local_checkpoint != verified_checkpoint.data() {
+            let verified_contents = self
+                .get_checkpoint_contents(&verified_checkpoint.content_digest)
+                .map(|opt_contents| {
+                    opt_contents
+                        .map(|contents| format!("{:?}", contents))
+                        .unwrap_or_else(|| {
```

Comment:
This could happen because verified contents are written by state-sync which runs async to checkpoint builder. When that happens, we still want to log the local checkpoint content so that we could compare that with the real content that we could grab from other places.
So I think we should read the local content first, and if we cannot find verified content we should still print local content

Reply:
I don't understand - this code already prints everything that it can. there are no early returns

Reply:
Oh you are right! I misread